### PR TITLE
Arc-2308 add backend ffs

### PR DESCRIPTION
**Pull request by rachellerathbone**
**What's in this PR?**
Addition of feature flag module for backend.

**Why**
So we can make changes, roll them out, and not break shit!

**Affected issues**  
Arc-2308

**How has this been tested?**  
Locally and stg

**What's Next?**
Start using them 😄

### CODE REVIEW

File: app/.nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-16.20.2
\ No newline at end of file
+18.17.0
```
Comment (rachellerathbone) [inline app/.nvmrc:1]: Forge cli was complaining I was using an old version and was having issues running forge commands. Had to bump, which came parcelled with a need to change this.

File: app/src/config/env.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export interface EnvVars {
+    LAUNCHDARKLY_API_KEY: string;
+    LAUNCHDARKLY_APP_NAME: string;
+}
+
+const envVars: EnvVars = {
+    LAUNCHDARKLY_API_KEY: process.env.LAUNCHDARKLY_API_KEY || '',
```
Comment (rachellerathbone) [inline app/src/config/env.ts:7]: These have been set for the appId using some forge cli commands. For more info see https://developer.atlassian.com/platform/forge/environments-and-versions/

File: app/src/config/feature-flags.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+};
+
+const logger = Logger.getInstance('featureFlags');
+
+async function getFeatureFlag(featureFlagKey: string): Promise<FeatureFlag> {
+    const eventType = 'retrievingFeatureFlag';
+    const errorMsg = `fetching feature flag unexpected status`;
+
+    try {
+        const response = await fetch(`${BASE_URL}/${featureFlagKey}`, { ...baseHeaders });
```
Comment (rachellerathbone) [inline app/src/config/feature-flags.ts:104]: Using fetch to call the LD API as we currently can't use the @launchdarkly/node-server-sdk due to Forge's lack of support for the Node.js runtime. https://hello.atlassian.net/wiki/spaces/OTFS/pages/2803689031/Forge+Limitations Work is being done to add this support and we can opt in by being onboarded to a feature flag but doing some removed the ability to run `forge tunnel` locally... Figured this was our next best option.

File: app/src/config/feature-flags.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    headers: {
+        Authorization: envVars.LAUNCHDARKLY_API_KEY
+    }
+};
+
+const logger = Logger.getInstance('featureFlags');
+
+async function getFeatureFlag(featureFlagKey: string): Promise<FeatureFlag> {
+    const eventType = 'retrievingFeatureFlag';
+    const errorMsg = `fetching feature flag unexpected status`;
```
Comment (joshkay10) [inline app/src/config/feature-flags.ts:99]: /nit ` without placeholder value

**Review by joshkay10 — APPROVED** [review]

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**